processor/src/main/java/org/mapstruct/ap/internal/model/source/SelectionParameters.java

```diff
@@ -0,0 +1,68 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.model.source;
+
+import java.util.List;
+import javax.lang.model.type.TypeMirror;
+
+/**
+ * Holding parameters common to the selection process, common to IterableMapping, BeanMapping, PropertyMapping and
+ * MapMapping
+ *
+ * @author Sjaak Derksen
+ */
+public class SelectionParameters {
```

Comment:
I thought about that as well. However:
1. Selection Criteria contain more criteria, e.g. the ones that are derived during the process.
2. Package restrictions restrict us from global usage. It feels like a mixup of concerns to move SectionCriteria to the source package (away from the selection process).
However, bundling selection parameters is a logical thing to do. I spotted a fault, because in `Mapping` they should for instance not be part of the reverse mapping.